jx/build/JXMLCompiler.js

```diff
@@ -152,7 +154,7 @@ JXMLCompiler.prototype.extractScript = function(build_assets) {
 			.replace(/\$path\b/g, build_assets.IDs[id]);
 
 
-	var FUNCTION_REGEX = /function +([$A-Z_][0-9A-Z_$]*)/i;
+	var FUNCTION_REGEX = /function +([$A-Z_][0-9A-Z_$]*)/gi;
```

Comment:
suppose I have the code like this

```javascript
<script>
function getNumber() {
  // do something with count() ...
  function count() {};  // private function
}
</script>
```
this will extract out `count`  and put it under `this.locals`, creating an error `count is undefined`

```javascript
this.locals['count'] = count;  // -> Error: count is undefined

function getNumber() {
  // do something with count() ...
  function count() {}; // private function 
}
```

So, should we make a exposed casting handler explicit ?

```javascript
handle_cast = function(m, a) { this.cast_handlers[m] && this.cast_handlers[m].apply(this, a) }

<script>
self.cast_handlers.getNumber = function() {};
</script>
```